### PR TITLE
MULE-12377: Review transform OperationModel

### PR DESCRIPTION
**Pull request by pablomusumeci**

### CODE REVIEW

File: modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/declaration/DefaultXmlArtifactDeclarationLoader.java [hunk 1]
```diff
@@ -108,6 +110,9 @@
  */
 public class DefaultXmlArtifactDeclarationLoader implements XmlArtifactDeclarationLoader {
 
+  public static final String SET_VARIABLE = "setVariable";
```
Comment (pablomusumeci) [inline modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/declaration/DefaultXmlArtifactDeclarationLoader.java]: This doesn't seem like the right place for this constants. Suggestions are welcomed.

File: modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/model/internal/DeclarationBasedElementModelFactory.java [hunk 2]
```diff
@@ -382,24 +385,26 @@ private void addAllDeclaredParameters(List<ParameterModel> parameterModels,
...
-      String missing = parameterDeclarations.stream()
-          .filter(p -> !configuredParameters.contains(p.getName()))
-          .map(ElementDeclaration::getName)
-          .collect(joining(","));
-
-      throw new IllegalArgumentException(
-                                         format("The parameters [%s] were declared but they do not exist in the associated model",
-                                                missing));
-    }
+    // TODO VALIDATE
```
Comment (pablomusumeci) [inline modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/model/internal/DeclarationBasedElementModelFactory.java]: Forgot to change this

Comment (marianogonzalez) [inline modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/model/internal/DeclarationBasedElementModelFactory.java]: remove

File: modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/model/internal/DeclarationBasedElementModelFactory.java [hunk 3]
```diff
@@ -415,8 +420,34 @@ private void addAllDeclaredParameters(List<ParameterModel> parameterModels,
...
+
+                  if (parameterValue != null) {
+                    groupDsl.getContainedElement(p.getName())
+                        .ifPresent(paramDsl -> {
+                          addParameter(p.getName(), parameterValue, p, paramDsl, groupBuilder, groupElementBuilder);
+                        });
+                  }
+
+                });
+
```
Comment (marianogonzalez) [inline modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/model/internal/DeclarationBasedElementModelFactory.java]: empty lines

File: modules/extensions-support/src/main/resources/META-INF/mule-extension-model.json [hunk 4]
```diff
@@ -976,6 +1123,188 @@
...
+                  "typeAlias": {
+                    "value": "Object"
+                  }
+                },
+                "fields": []
+              },
+              "hasDynamicType": false,
+              "required": false,
+              "isConfigOverride": false,
+              "expressionSupport": "NOT_SUPPORTED",
```
Comment (alegmarra) [inline modules/extensions-support/src/main/resources/META-INF/mule-extension-model.json]: REQUIRED

File: modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/declaration/DefaultXmlArtifactDeclarationLoader.java (hunk 1 above)
Comment (alegmarra) [inline modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/declaration/DefaultXmlArtifactDeclarationLoader.java]: Is fine for the hack, use "TRANSFORMER_" as prefix for the var name. Move TRANSFORM_IDENTIFIER out of DslConstansts and put it here too.

File: modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/model/internal/DeclarationBasedElementModelFactory.java (hunk 2 above)
Comment (alegmarra) [inline modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/model/internal/DeclarationBasedElementModelFactory.java]: remove

File: modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/model/internal/DeclarationBasedElementModelFactory.java [hunk 5]
```diff
@@ -768,16 +799,31 @@ private void createList(ParameterListValue list,
...
+        listDsl.getGeneric(itemType).isPresent() ? listDsl.getGeneric(itemType).get() : createSetVariableDslElement();
+    list.getValues().forEach(value -> createListItemConfig(itemType, value, itemDsl, listConfig, listElement));
     ComponentConfiguration result = listConfig.build();
 
+
     parentConfig.withNestedComponent(result);
     parentElement.containing(listElement.withConfig(result).build());
   }
 
+  private DslElementSyntax createSetVariableDslElement() {
```
Comment (alegmarra) [inline modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/model/internal/DeclarationBasedElementModelFactory.java]: `createTransformSetVariableParameterDslElement`... if it's hackish, let's be clear about it

**Pull request merged**